vowpalwabbit/memory_tree.cc

```diff
@@ -1259,8 +1260,9 @@ base_learner* memory_tree_setup(options_i& options, vw& all)
                .default_value(0)
                .help("turn on dream operations at reward based update as well"))
       .add(make_option("online", tree->online).help("turn on dream operations at reward based update as well"));
-  options.add_and_parse(new_options);
-  if (!tree->max_nodes)
+
+  // TODO: check this one
+  if (!options.add_parse_and_check_necessary(new_options))
```

Comment:
Should there be an additional check for 0?

Reply:
I don't think so, but left a comment to verify that the behavior holds up.

Reply:
done and verified